syntax = "proto3";

package reasonchip;

service ReasonChipService {

    // Bidirectional streaming RPC
    rpc EstablishConnection (stream ReasonChipPacket) returns (stream ReasonChipPacket);

}

message UUID {
	string value = 1;
}

message ReasonChipPacket {
	string packet_type = 1;
	UUID cookie = 2;
	string pipeline = 3;
	string variables = 4;
	string rc = 5;
	string error = 6;
	repeated string stacktrace = 7;
	string result = 8;
}

